src/bindgen/ir/enumeration.rs

```diff
@@ -142,7 +143,10 @@ impl EnumVariant {
         }
 
         let variant_cfg = Cfg::append(mod_cfg, Cfg::load(&variant.attrs));
-        let annotations = AnnotationSet::load(&variant.attrs)?;
+        let mut annotations = AnnotationSet::load(&variant.attrs)?;
+        if let Some(b) = enum_annotations.bool("derive-ostream") {
+            annotations.add("derive-ostream", AnnotationValue::Bool(b));
```

Comment:
Maybe this should only add if the annotation is not present on the variant?

Reply:
Just to enumerate all the cases as I thought through this:
- in the general case, if the enum has `derive-ostream=true` and one of the variant bodies has `derive-ostream=false`, that's not going to compile because the serializer function generated for the enum will try to use the serializer function for the variant body, which won't exist. I guess the user could define it somewhere else, and so we could allow it on those grounds. If they're explicitly setting the annotation on the variant body then we can assume they know what they're doing.
- the rest of the cases below refer to the other interesting scenario, where the enum has `derive-ostream=false` and the variant inside has `derive-ostream=true`
-- the untagged enum case, all the variants are going to be of type `VariantBody::Empty`, and we don't even read the derive-ostream annotation from those. So yes, I can avoid adding the annotation in that case (or for all `VariantBody::Empty` cases in general)
-- the tagged enum case with `separate_tag = true`: in this case we can generate the serializer for the struct that represents the variant body, regardless of whether or not the enclosing enum has a serializer. So in this case, yes, I can avoid propagating the annotation down
-- the tagged enum case with `separate_tag = false`: in this case the variant body contains a field that is the tag, so if we are generating a serializer for the variant body, then we need the tag serialized as well, otherwise there will be a compilation failure. But that scenario is actually already broken in the current patch if the enum doesn't explicitly set `derive-ostream=false` but instead just has a default of false. So I guess this scenario is not made any worse. And again, this is a scenario where the user did not set the annotation on the enum but instead on a specific variant body, so maybe we can assume they know what they're doing.

So yes overall your suggestion makes sense. I'll submit another PR with a patch.

Reply:
https://github.com/eqrion/cbindgen/pull/585